### ISSUE TITLE
ImGuiSelectableFlags_SpanAllColumns do not span whole table if ImGuiTableFlags_ScrollFreezeLeftColumn set and scrolled horizontally to the right

### ISSUE THREAD

**Issue by jpbrault** [ocornut/imgui#3386]
**Version/Branch of Dear ImGui:**

Version: 1.78 WIP
Branch: table

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_sdl.cpp + imgui_impl_opengl2.cpp
Compiler: GCC 9.1 / Clang 10
Operating System: Mac/Linux

**My Issue/Question:**

When using a Selectable() in a table row with the ImGuiSelectableFlags_SpanAllColumns flag in a table with ImGuiTableFlags_ScrollFreezeTopRow | ImGuiTableFlags_Scroll | ImGuiTableFlags_ScrollFreezeLeftColumn flags, I would expect the selection to extent past the last column so the whole row would be selected/highlighted. I would also expect that column highlight to header to align with the cell.

**Screenshots/Video**

![scroll1](https://user-images.githubusercontent.com/8876173/89429297-6659e180-d70b-11ea-8fef-24f28eba8e5a.png)
![scroll2](https://user-images.githubusercontent.com/8876173/89429310-68bc3b80-d70b-11ea-9524-858d2e6230b4.png)
![scroll3](https://user-images.githubusercontent.com/8876173/89429322-6b1e9580-d70b-11ea-982e-eafa4d21782a.png)
![scroll3](https://user-images.githubusercontent.com/8876173/89429336-6e198600-d70b-11ea-909d-be19a0b45624.png)
![scroll4](https://user-images.githubusercontent.com/8876173/89429339-6e198600-d70b-11ea-91d5-5f89b029e4e7.png)
![scroll5](https://user-images.githubusercontent.com/8876173/89429346-6fe34980-d70b-11ea-8759-4eceb74672f9.png)
![scroll6](https://user-images.githubusercontent.com/8876173/89429353-7245a380-d70b-11ea-862d-3277cbd635b1.png)

```
  if ( ImGui::Begin( "Window" ) )
  {
    if ( ImGui::BeginTable(
         "Table", 2,
            ImGuiTableFlags_ScrollFreezeTopRow |
           ImGuiTableFlags_Scroll |
           ImGuiTableFlags_ScrollFreezeLeftColumn ) )
    {
      ImGui::TableSetupColumn(
        "Col 1", ImGuiTableColumnFlags_WidthFixed | ImGuiTableColumnFlags_NoHide,
        200 );
      ImGui::TableSetupColumn(
        "Col 2", ImGuiTableColumnFlags_WidthFixed,
        200 );
      ImGui::TableAutoHeaders();

      ImGui::TableNextRow();
      ImGui::TableSetColumnIndex( 0 );

      ImGui::Selectable( "Val 1", true,
                         ImGuiSelectableFlags_SpanAllColumns |
                          ImGuiSelectableFlags_AllowDoubleClick );
      if ( ImGui::TableNextCell() )
        ImGui::TextUnformatted( "Val 2");

      ImGui::EndTable();
    }
    ImGui::End();
  }
```

**Comment by ocornut** [comment]
Hello and thanks for the report.
I believe this is merely a variant of the issue described in #3187.
Going to look at that more seriously.

**Comment by ocornut** [comment]
This should now be fixed in master by 8074b4914834e2892000fc1f359fdb8cf84e88ac and in tables with 8074b4914834e2892000fc1f359fdb8cf84e88ac.

Thanks for your help and report!
